lib/mpool/src/mdc_file.c

```diff
@@ -598,7 +598,7 @@ mdc_file_read(struct mdc_file *mfp, void *data, size_t len, bool verify, size_t
     int               rhlen, rc;
     merr_t            err;
 
-    if (!mfp || !data)
+    if (!mfp || (len && !data))
```

Comment:
This enables getting record length with `mdc_file_read(mfp, NULL, 0, &recsize);`

Reply:
Is returning EOVERFLOW fine for this use-case?

Reply:
mdc_file_read() returns EOVERFLOW if the size of the input buffer is smaller than the to-be read MDC record size and fills in the actual record size in the output param 'recsize' so that the user can pass in an appropriately sized buffer the next time.

Alex has added a special case where a client can just retrieve the MDC record size by not passing a buffer (!data && len == 0). The code currently fills the record size and returns EOVERFLOW for this case as well. I was wondering whether the API should return success for this special case as there is no buffer passed by the client to overflow.